src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -765,7 +767,15 @@ private IDynamicInvocationOperation CreateBoundDynamicCollectionElementInitializ
             ITypeSymbol type = boundCollectionElementInitializer.Type;
             Optional<object> constantValue = ConvertToOptional(boundCollectionElementInitializer.ConstantValue);
             bool isImplicit = boundCollectionElementInitializer.WasCompilerGenerated;
-            return new LazyDynamicInvocationExpression(operation, arguments, argumentNames: ImmutableArray<string>.Empty, argumentRefKinds: ImmutableArray<RefKind>.Empty, _semanticModel, syntax, type, constantValue, isImplicit);
+            BoundImplicitReceiver implicitReceiver = boundCollectionElementInitializer.ImplicitReceiver;
+            Lazy<IOperation> addReference = new Lazy<IOperation>(() => CreateBoundDynamicMemberAccessOperation(implicitReceiver,
+                                                                                                               typeArgumentsOpt: ImmutableArray<TypeSymbol>.Empty,
+                                                                                                               memberName: "Add",
+                                                                                                               implicitReceiver.Syntax,
+                                                                                                               implicitReceiver.Type,
```

Comment:
>implicitReceiver.Type [](start = 111, length = 21)

I think type should be null here, it is definitely not the same as the type of the receiver

Reply:
For consistency with other DynamicMemberReferences, I'll make this the DynamicType

---
In reply to: [192265008](https://github.com/dotnet/roslyn/pull/27308#discussion_r192265008) [](ancestors = 192265008)

Reply:
> For consistency with other DynamicMemberReferences, I'll make this the DynamicType

That would make sense only if there is a separate dynamic invocation for this node, i.e. two dynamic invocations in IL rather than one. Please check IL, and if there is only one dynamic invocation, the type should be null because this is not a value from compiler's point of view

---
In reply to: [192269007](https://github.com/dotnet/roslyn/pull/27308#discussion_r192269007) [](ancestors = 192269007,192265008)